paymentsheet/src/main/java/com/stripe/android/customersheet/StripeCustomerAdapter.kt

```diff
@@ -112,15 +121,25 @@ internal class StripeCustomerAdapter @Inject constructor(
         }
     }
 
-    override suspend fun setupIntentClientSecretForCustomerAttach(): Result<String> {
-        return getCustomerEphemeralKey().mapCatching { customerEphemeralKey ->
-            setupIntentClientSecretProvider?.provideSetupIntentClientSecret(customerEphemeralKey.customerId)
-        }.getOrElse {
-            Result.failure(it)
-        } ?: throw IllegalArgumentException("setupIntentClientSecretProvider cannot be null")
+    override suspend fun setupIntentClientSecretForCustomerAttach(): CustomerAdapter.Result<String> {
+        if (setupIntentClientSecretProvider == null) {
+            throw IllegalArgumentException("setupIntentClientSecretProvider cannot be null")
+        }
+        return getCustomerEphemeralKey()
+            .mapCatching { customerEphemeralKey ->
+                setupIntentClientSecretProvider.provideSetupIntentClientSecret(customerEphemeralKey.customerId)
+            }.fold(
+                onSuccess = { it },
+                onFailure = { cause, displayMessage ->
```

Comment:
What are we mapping this from? How come they're not the same types?

Reply:
I noticed this last week too. I am addressing your concern [here](https://github.com/stripe/stripe-android/pull/6993).